### ISSUE TITLE
Captions for images

### ISSUE THREAD

**Issue by jkapusci** [sul-cidr/noh#319]
These can perhaps be hidden under an Info icon just like it is now for video or more traditionally appear in small font below the image.

**Comment by csbailey5t** [comment]
We need a caption template with html5 caption tag. We prefer visible under image or video or sound.

**Comment by broadwell** [comment]
`staging/props/` is an example of a page with multiple image captions, currently implemented via `<h7>` tags, that probably should be accomplished via HTML5 `<figure>` and `<figcaption>` elements instead.

**Comment by jkapusci** [comment]
Dear Peter,
Thank you for the great suggestion. I just tried to switch and it works nicely although we lost the top and bottom padding for the image that was coded in the class="image-container--no-background”
Can we preserve the padding somehow? It would be great also that the figcaption matches the font of the rest just make it smaller. What do you think?
Would you like to look at the include/image-no-background.html to see what is it doing exactly and if you can amend the css perhaps? Thanks!
Jarek

***
Jaroslaw Kapuscinski
Associate Professor
Chair, Department of Music
Stanford University
tel. (650) 723 2717

On Jul 22, 2019, at 11:53 AM, Peter Broadwell <notifications@github.com<mailto:notifications@github.com>> wrote:


staging/props/ is an example of a page with multiple image captions, currently implemented via <h7> tags, that probably should be accomplished via HTML5 <figure> and <figcaption> elements instead.

**Comment by broadwell** [comment]
Dear @jkapusci,
Thanks for the quick feedback! I'm pretty sure that the top and bottom padding of the images should still be there; what is supposed to have gone away is the padding between the image and the caption, because by default they're in the same layout block. But it's easy enough to add some padding there, as I've just done in the newest update to the `feature/image-captions` branch. I also adjusted the styling of the caption text so that it's similar to the rest of the page but smaller, as you suggested. I can of course continue to modify all of these aspects, so please check out the changes if possible and let me know what you think.
Peter

**Comment by jkapusci** [comment]
Dear Peter,
I just pushed a version of the page with top image using figcaption and bottom the include/image.hml.
I am curious to see it and its padding remotely.
What would it take for me to see it locally?
Thanks!
Jarek

***
Jaroslaw Kapuscinski
Associate Professor
Chair, Department of Music
Stanford University
tel. (650) 723 2717

On Jul 22, 2019, at 3:42 PM, Peter Broadwell <notifications@github.com<mailto:notifications@github.com>> wrote:


Dear @jkapusci<https://github.com/jkapusci>,
Thanks for the quick feedback! I'm pretty sure that the top and bottom padding of the images should still be there; what is supposed to have gone away is the padding between the image and the caption, because by default they're in the same layout block. But it's easy enough to add some padding there, as I've just done in the newest update to the feature/image-captions branch. I also adjusted the styling of the caption text so that it's similar to the rest of the page but smaller, as you suggested. I can of course continue to modify all of these aspects, so please check out the changes if possible and let me know what you think.
Peter

**Comment by broadwell** [comment]
Dear @jkapusci,
Ah, I see -- the remote version is indeed almost in accord with the suggested changes; the main bit missing, as you noted, is the `class="image-container--no-background"` attribute of the `<figure>` tag, which is responsible for bringing in the CSS padding settings. In practice, though, when using the new branch, all you'd really need to do is to follow the modified `image-no-background.html` template, which I'll eventually add to the wiki. It would then expand to produce a properly formatted figure+figcaption entry. So for example, you'd replace the lines for the first image and caption in `props.md` with the following:
```
{% include image-no-background.html src="/assets/images/Okura-dwelling.png" caption="Dwelling property for Hashitomi. Ōkura ryū tsukurimono hiroku (Ōkura School Secret Records on Stage Properties) by Yamamura Kyūzaemon. Noh Theater Research Institute of Hōsei University. Available at  <a href=\"http://www.dh-jac.net/db1/books/results1024.php?f1=nohken-y17-29&f12=1&enter=jparc&skip=17&-max=1&enter=jparc\">JPARC</a> or <a href=\"https://nohken.ws.hosei.ac.jp/nohken_material/htmls/index/pages/y17/29.html\">Hōsei University.</a>" %}
```
So if you pull in the changes from the `feature/image-captions` branch locally, update `props.md` as suggested above, and then run `yarn dev`, you should be able to see the first image at `http://127.0.0.1:4000/staging/props/` with a properly formatted caption. If you'd rather not deal with git's bizarre customs, however, you can also just check out my demo site here:
http://ec2-3-14-12-54.us-east-2.compute.amazonaws.com/staging/props/

Please let me know what you think of the layout and formatting of the first captioned image! Cheers,
Peter

**Comment by jkapusci** [comment]
Dear Peter,
Brilliant. Thank you! I dared to toy with github and am delighted with the result. I have updated both images and pushed the file.
Thanks again!
Jarek

***
Jaroslaw Kapuscinski
Associate Professor
Chair, Department of Music
Stanford University
tel. (650) 723 2717

On Jul 22, 2019, at 10:02 PM, Peter Broadwell <notifications@github.com<mailto:notifications@github.com>> wrote:


Dear @jkapusci<https://github.com/jkapusci>,
Ah, I see -- the remote version is indeed almost in accord with the suggested changes; the main bit missing, as you noted, is the class="image-container--no-background" attribute of the <figure> tag, which is responsible for bringing in the CSS padding settings. In practice, though, when using the new branch, all you'd really need to do is to follow the modified image-no-background.html template, which I'll eventually add to the wiki. It would then expand to produce a properly formatted figure+figcaption entry. So for example, you'd replace the lines for the first image and caption in props.md with the following:

{% include image-no-background.html src="/assets/images/Okura-dwelling.png" caption="Dwelling property for Hashitomi. Ōkura ryū tsukurimono hiroku (Ōkura School Secret Records on Stage Properties) by Yamamura Kyūzaemon. Noh Theater Research Institute of Hōsei University. Available at  <a href=\"http://www.dh-jac.net/db1/books/results1024.php?f1=nohken-y17-29&f12=1&enter=jparc&skip=17&-max=1&enter=jparc\">JPARC</a> or <a href=\"https://nohken.ws.hosei.ac.jp/nohken_material/htmls/index/pages/y17/29.html\">Hōsei University.</a>" %}


So if you pull in the changes from the feature/image-captions branch locally, update props.md as suggested above, and then run yarn dev, you should be able to see the first image at http://127.0.0.1:4000/staging/props/ with a properly formatted caption. If you'd rather not deal with git's bizarre customs, however, you can also just check out my demo site here:
http://ec2-3-14-12-54.us-east-2.compute.amazonaws.com/staging/props/

Please let me know what you think of the layout and formatting of the first captioned image! Cheers,
Peter

**Comment by broadwell** [comment]
Dear @jkapusci,
Now that Scott has merged the captions code into the `develop` branch, you should be able to see the caption functionality locally by updating that branch and then following the above suggestions for modifying the `{% include image-no-background.html ...` entries to include the caption text.
I've also updated the image container templates in the wiki to show where the captions should go and how they need to be formatted if they contain quotes:
https://github.com/sul-cidr/noh/wiki/List-of-HTML-components

Peter

**Comment by csbailey5t** [comment]
Closed in #371
